sns_sqs_communicator/queue/protocol.py

```diff
@@ -0,0 +1,58 @@
+import collections.abc
+import typing
+
+import mypy_boto3_sqs.type_defs
+
+from .. import clients, fifo_attributes_creator
+
+
+class QueueProtocol(typing.Protocol):
+    """Protocol for queue."""
+
+    client: clients.SQSClient
+    queue_url: str
+    fifo_attrs_creator: (
+        fifo_attributes_creator.FifoAttributesCreatorProtocol | None
+    ) = None
+    wait_time_seconds: int = 0
+
+    def __init__(
```

Comment:
Maybe it's better to move constructor to concrete `SQSQueue` class? Or get rid of protocol then, if we don't need to support other types of queues. Because this protocol's purpose was to have an abstract layer on top of any queue, so it should be independent of any code that is related to sqs